/*******************************************************************************
    Copyright (C) 2014-2016 Wright State University - All Rights Reserved
    Daniel P. Foose - Maintainer/Lead Developer

    This file is part of Vespucci.

    Vespucci is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    Vespucci is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Vespucci.  If not, see <http://www.gnu.org/licenses/>.
*******************************************************************************/
#ifndef MAPDIALOG_H
#define MAPDIALOG_H

#include <QDialog>
#include "Global/vespucciworkspace.h"
class MainWindow;
namespace Ui {
class MapDialog;
}

class MapDialog : public QDialog
{
    Q_OBJECT

public:
    explicit MapDialog(MainWindow *parent, QStringList data_keys, QSharedPointer<VespucciWorkspace> ws);
    ~MapDialog();

private slots:
    void on_buttonBox_accepted();

    void on_columnSpinBox_valueChanged(int arg1);

private:
    Ui::MapDialog *ui;
    MainWindow *main_window_;
    QSharedPointer<VespucciWorkspace> workspace_;
    QSharedPointer<VespucciDataset> dataset_;
    QStringList data_keys_;
};

#endif // MAPDIALOG_H
